utilities/general-helpers.metta

```diff
@@ -47,3 +47,27 @@
 )
 )
 )
+
+; This function assumes it receives a set1 and set2 expression as parameters. 
+; It then removes similar elements found in set2 that already exist in set1.
+; The function uses isMember from general helpers functions.
+
+(: setDifference (-> Expression Expression Expression))
+(= (setDifference $set1 $set2) 
+    (if (== $set1 ()) () ; Base case: return empty set when $set1 is empty
+        (let* (
+            ($head (car-atom $set1))  ; Extract the head of set1
+            ($tail (cdr-atom $set1))  ; Extract the tail of set1
+            ($newtail (setDifference $tail $set2))  ; Recursively process the tail
+        )
+            ; Check if the $head is in set2, if so, skip it
+
+            (if (and (isMember $head $set2) (not (== (get-metatype $head) Expression)) )
```

Comment:
Hey @YeabsiraNigusse, why do we need to make sure that the `$head` isn't an expression? Shouldn't the set difference work with any type of atom as an element in the set to be considered generic?

Reply:
@Yagth . Yes you are right! i was considering one RTE function case but yeah if it is a utility function it should be generic. i pushed another change. you can take a look.